### ISSUE TITLE
Mixed comment problem afer 1.9.9

### ISSUE THREAD

**Issue by tsherr** [nojanath/SublimeKSP#162]
Somewhere after 1.9.9, comments of the form `//******/` no longer compile.

I suspect it has something to do with the unintended mixed use of `//` and `/*`, but the intent here is stylistic and perfectly legal.

A more practical example follows:
```
//****************SCRIPT TITLE****************
/*
 * Detailed notes
 * Blah blah
 */
```

**Comment by pryzmusic** [comment]
Hi,

I've tested this code and can not replicate the issue. I am using version 1.9.11
```
//****************SCRIPT TITLE****************
/*
 * Detailed notes
 * Blah blah
 */

 on init
 	declare a
 	message(a)
 end on

 { Compiled on Thu Jun 18 14:36:59 2020 }
on init
declare $a
message($a)
end on
```

Can you provide a self-contained piece of code that I can test?

**Comment by tsherr** [comment]
Interesting, I just tested the exact example you posted and it doesn't work for me. I've tried it with different compiler options too (extra syntax checks, etc.).

Each time, I get the following error:
```
Syntax error (line 0)
/
<main script>:1
```

Using v1.9.11 of the plugin
Sublime 3.2.2
Windows 10

**Comment by pryzmusic** [comment]
Hi, I've tested it again today, and now it doesn't work. This is actually a good thing, because logically speaking, it SHOULD break the way the compiler is currently configured.

Unfortunately, the way // comments were introduced into the compiler makes it impossible to properly support both types without evaluating them in a specific order. I switched the order so that closed comments like { } would take precedence over open comments (//) so something like

```
{ // something // }
```

would not return an error. This would mean that now closed comments halfway inside of open comments will now operate as comments which is the issue you are describing.

The only thing I can recommend for now is adding a space between the `//` and `*****...` for now in your first line. The proper solution that would evaluate comments correctly no matter the order of appearance will require a complete overhaul of the part of the code that evaluates them, and such an operation is incredibly risky as rewriting compiler functions can break any manner of edge cases that KSP developers rely on since there is no central style guide for writing KSP.

I will leave this as a request to fix but I can not devote the time to it for a little while.
